### ISSUE TITLE
App configuration as inline metadata (toml comment)?

### ISSUE THREAD

**Issue by manzt** [marimo-team/meps#7]
First of all, I want to say that I really _love_ the MEPs. These are excellent reads and really give me a sense of the foundations and considerations that have gone into marimo. Kudos!

I saw in [MEP 4](https://github.com/marimo-team/meps/blob/main/mep-0004.md) that the idea for putting the app config as TOML was ultimately decided against because it:

> "requires committing and shipping the TOML file for settings to be reproducible. This was a dealbreaker."

I'm guessing that this decision was before PEP 723, and now there is some precedent for putting tool-specific metadata in the comment block as well. For example, with uv you can add `exclude-newer` metadata.

```python
# /// script
# requires-python = ">=3.13"
# dependencies = [
#     "numpy",
# ]
#
# [tool.uv]
# exclude-newer = "2024-12-05T22:06:45.27544-05:00"
# ///

def main() -> None:
    print("Hello from foo.py!")


if __name__ == "__main__":
    main()
```

I'm not saying this _should_ be where marimo app configuration lives, but moreover just pointing out that the original con/dealbreaker wrt to more static app configuration may no longer hold. Thought I'd open an issue for discussion.

**Comment by mscolnick** [comment]
Yes, we definitely now ship toml with our notebooks (only optionally), for package requirements.

Good timing on this issue/topic though, since I am adding `[tool.marimo]` support to the local `pyproject.toml`: https://github.com/marimo-team/marimo/pull/3071

I was planning in a followup to also read from the current file for additional user-config overrides as well, and they all merge back up. You can probably see where this would go (and how easy it is):

```python
def get_default_config_manager(*, current_path: Optional[str]) -> MarimoConfigManager:
    # Current path should be the notebook file
    # If it's not known, use the current working directory
    if current_path is None:
        current_path = os.getcwd()
    return MarimoConfigManager(
        UserConfigManager(),
        ProjectConfigManager(current_path),
        # add more overrides here
    )
```

Note: this would be for user-configuration (theme, display options, runtime, etc). and not the existing app-configuration (title, layout).

I am not sure how these evolve overtime, but for now, i think it makes sense to allow user-configs (e.g. `[tool.marimo]`) to be set in marimo files in the same vain of "self-containment"
